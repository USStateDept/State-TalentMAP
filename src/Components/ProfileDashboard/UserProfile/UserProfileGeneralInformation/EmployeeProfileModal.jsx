import PropTypes from 'prop-types';
import swal from '@sweetalert/with-react';
import { Viewer, Worker } from '@react-pdf-viewer/core';
import { fetchJWT, getAssetPath } from 'utilities';
import { defaultLayoutPlugin } from '@react-pdf-viewer/default-layout';

// Import styles
import '@react-pdf-viewer/core/lib/styles/index.css';
import '@react-pdf-viewer/default-layout/lib/styles/index.css';

const EmployeeProfileModal = props => {
  const { url } = props;

  const cancel = (e) => {
    e.preventDefault();
    try {
      swal.close();
    } catch {
      return null;
    }
    return null;
  };

  const defaultLayoutPluginInstance = defaultLayoutPlugin();

  return (
    <div className="checklist-modal">
      <div>
        <Worker workerUrl={getAssetPath('/static/js/pdf.worker.js')}>
          {/* <Viewer fileUrl={this.state.data} />; */}
          <Viewer
            fileUrl={url}
            httpHeaders={{ JWTAuthorization: fetchJWT() }}
            withCredentials
            plugins={[defaultLayoutPluginInstance]}
          />
        </Worker>
        <div className="checklist-modal-buttons-container">
          <button type="button" onClick={cancel}>Close</button>
        </div>
      </div>
    </div>
  );
};

EmployeeProfileModal.propTypes = {
  url: PropTypes.string,
};

EmployeeProfileModal.defaultProps = {
  url: '',
};

export default EmployeeProfileModal;
