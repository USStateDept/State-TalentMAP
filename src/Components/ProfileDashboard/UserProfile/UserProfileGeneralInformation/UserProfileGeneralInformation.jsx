import React from 'react';
import PropTypes from 'prop-types';
import { USER_PROFILE } from '../../../../Constants/PropTypes';
import SectionTitle from '../../SectionTitle';
import InformationDataPoint from '../../InformationDataPoint';
import ProfilePicture from '../../../ProfilePicture';
import StaticDevContent from '../../../StaticDevContent';
import SkillCodeList from '../../../SkillCodeList';

const UserProfileGeneralInformation = ({ userProfile, useGroup }) => (
  <div className="current-user-top current-user-section-container">
    <div className="section-padded-inner-container">
      <ProfilePicture />
      <SectionTitle small title={`${userProfile.user.last_name}, ${userProfile.user.first_name}`} className="current-user-name" />
      {
        useGroup ?
          <StaticDevContent>
            <InformationDataPoint
              content="Generalist â€¢ F2"
              className="skill-code-data-point-container skill-code-data-point-container-gen-spec"
            />
          </StaticDevContent>
          :
          <InformationDataPoint
            content={<SkillCodeList skillCodes={userProfile.skills} />}
            className="skill-code-data-point-container skill-code-data-point-container-skill"
          />
      }
    </div>
  </div>
);

UserProfileGeneralInformation.propTypes = {
  userProfile: USER_PROFILE.isRequired,
  useGroup: PropTypes.bool,
};

UserProfileGeneralInformation.defaultProps = {
  useGroup: false,
};

export default UserProfileGeneralInformation;
