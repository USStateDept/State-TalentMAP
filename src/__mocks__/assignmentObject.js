const assignmentObject = {
  start_date: '10-10-2016',
  end_date: '10-10-2018',
  estimated_end_date: '10-10-2018',
  position: {
    skill: 'CONSTRUCTION ENGINEERING (6218)',
    bureau: '(NEA) BUREAU NEAR EASTERN AFFAIRS',
    position_number: '58348065',
    title: 'PROJECT DIRECTOR',
    post: {
      location: {
        id: 4,
        country: 'United States',
        code: '500',
        city: 'Washington',
        state: 'DC',
      },
    },
  },
};

export default assignmentObject;
