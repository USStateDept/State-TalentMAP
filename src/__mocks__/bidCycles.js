export default [
  {
    id: 1,
    name: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
    cycle_start_date: '2017-12-13T15:51:10.980554Z',
    cycle_deadline_date: '2018-04-13T15:51:10.980554Z',
    cycle_end_date: '2018-06-13T15:51:10.980554Z',
    active: true,
  },
];
