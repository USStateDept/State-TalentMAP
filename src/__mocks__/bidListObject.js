const bidListObject = {
  count: 3,
  next: null,
  previous: null,
  results: [
    {
      id: 3,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 226,
        grade: '03',
        skill: 'POLITICAL AND ECONOMIC AFFAIRS (6050)',
        position_number: '10035561',
        title: 'POLITICAL/ECONOMIC OFFICER',
        create_date: '2015-08-05',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 199,
          location: 'Curacao',
        },
      },
      status: 'approved',
      submission_date: '2017-10-31',
      update_date: '2017-10-31',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 1,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 1,
        grade: '05',
        skill: 'OFFICE MANAGEMENT (9017)',
        position_number: '00003026',
        title: 'OMS (COM)',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 235,
          location: 'Freetown, Sierra Leone',
        },
      },
      status: 'closed',
      submission_date: '2017-10-31',
      update_date: '2017-11-01',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 2,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 84,
        grade: '04',
        skill: 'SECURITY (2501)',
        position_number: '56001600',
        title: 'SPECIAL AGENT',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 195,
          location: 'The Hague, Netherlands',
        },
      },
      status: 'closed',
      submission_date: null,
      update_date: '2017-11-01',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
  ],
};

export default bidListObject;
