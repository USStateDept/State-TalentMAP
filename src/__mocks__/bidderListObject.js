const bidderListObject = {
  count: 6,
  next: null,
  previous: null,
  results: [
    {
      id: 5,
      current_assignment: '(active) Kara Batisak at [58312265] PROJECT DIRECTOR (Guatemala, Guatemala)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '01',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 1,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Kara Batisak',
          draft: 6,
          submitted: 3,
          handshake_offered: 1,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'batisak',
        email: 'batisak@state.gov',
        first_name: 'Kara',
        last_name: 'Batisak',
      },
      language_qualifications: [],
    },
    {
      id: 3,
      current_assignment: '(active) John Doe at [10181012] INTERNATIONAL RELATIONS OFF (Singapore, Singapore)',
      skills: [
        {
          id: 55,
          cone: null,
          code: '6080',
          description: 'INTERFUNCTIONAL OFFICER OPER',
        },
      ],
      grade: '02',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'doej',
        email: 'doej@talentmap.us',
        first_name: 'John',
        last_name: 'Doe',
      },
      language_qualifications: [],
    },
    {
      id: 1,
      current_assignment: '(active) Guest McGuestson at [58312870] CONSTRUCTION MANAGER (Nogales, Mexico)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '03',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'guest',
        email: 'guest@state.gov',
        first_name: 'Guest',
        last_name: 'McGuestson',
      },
      language_qualifications: [],
    },
    {
      id: 6,
      current_assignment: '(active) Tarek Rehman at [56457009] ASST REGIONAL SECURITY OFCR (Adana, Turkey)',
      skills: [
        {
          id: 20,
          cone: null,
          code: '2501',
          description: 'SECURITY',
        },
      ],
      grade: '04',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 2,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Tarek Rehman',
          draft: 5,
          submitted: 2,
          handshake_offered: 3,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'rehmant',
        email: 'rehmant@state.gov',
        first_name: 'Tarek',
        last_name: 'Rehman',
      },
      language_qualifications: [],
    },
    {
      id: 2,
      current_assignment: '(active) Administrator TalentMAP at [58312869] PROJECT DIRECTOR (Nogales, Mexico)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '01',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'admin',
        email: 'admin@talentmap.us',
        first_name: 'Administrator',
        last_name: 'TalentMAP',
      },
      language_qualifications: [],
    },
    {
      id: 4,
      current_assignment: '(active) Jenny Townpost at [10026129] INTERNATIONAL RELATIONS OFF (Port Moresby, Papua New Guinea)',
      skills: [
        {
          id: 55,
          cone: null,
          code: '6080',
          description: 'INTERFUNCTIONAL OFFICER OPER',
        },
      ],
      grade: '02',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 4,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Jenny Townpost',
          draft: 3,
          submitted: 4,
          handshake_offered: 3,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'townpostj',
        email: 'townpostj@state.gov',
        first_name: 'Jenny',
        last_name: 'Townpost',
      },
      language_qualifications: [],
    },
  ],
};

export default bidderListObject;
