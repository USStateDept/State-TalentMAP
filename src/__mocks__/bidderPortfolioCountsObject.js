const bidderPortfolioCountsObject = {
  all_clients: 100,
  bidding_clients: 50,
  in_panel_clients: 20,
  on_post_clients: 30,
  bidding_no_handshake: 10,
};

export default bidderPortfolioCountsObject;
