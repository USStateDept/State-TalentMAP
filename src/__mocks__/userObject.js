export const bidderUserObject = {
  skill_code: ['INFORMATION MANAGEMENT (2880)', 'ADMINISTRATIVE SUPPORT (2080)'],
  user: {
    first_name: 'John',
    last_name: 'Doe',
    email: 'doej@state.gov',
    username: 'jdoe',
  },
  is_cdo: false,
  cdo: {
    email: 'shadtrachl@state.gov',
  },
  bid_statistics: [
    {
      draft: 1,
      submitted: 1,
    },
  ],
};

export const cdoUserObject = Object.assign({}, bidderUserObject, { is_cdo: true });
