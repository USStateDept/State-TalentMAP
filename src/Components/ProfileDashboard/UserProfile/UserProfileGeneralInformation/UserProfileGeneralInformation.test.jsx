import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { bidderUserObject } from '../../../../__mocks__/userObject';
import {
  testDispatchFunctions,
} from '../../../../testUtilities/testUtilities';
import UserProfileGeneralInformation, { mapDispatchToProps } from './UserProfileGeneralInformation';

describe('UserProfileGeneralInformationComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<UserProfileGeneralInformation.WrappedComponent
      userProfile={bidderUserObject}
    />);
    expect(wrapper).toBeDefined();
  });

  it('renders current user name', () => {
    const wrapper = shallow(<UserProfileGeneralInformation.WrappedComponent
      userProfile={bidderUserObject}
    />);
    expect(wrapper.find('.name-group').children().find('PositionInformation').at(0)
      .props().title).toBe('Doe, John');
  });

  it('renders Avatar', () => {
    const wrapper = shallow(<UserProfileGeneralInformation.WrappedComponent
      userProfile={bidderUserObject}
      colorProp="displayName"
      useColor
    />);
    expect(wrapper.find('Avatar').props().colorString).toBe('John');
    expect(wrapper.find('Avatar').props().initials).toBe('JD');
  });

  it('matches snapshot when showEditLink is true', () => {
    const wrapper = shallow(
      <UserProfileGeneralInformation.WrappedComponent
        showEditLink
        userProfile={bidderUserObject}
      />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when useGroup is true', () => {
    const wrapper = shallow(
      <UserProfileGeneralInformation.WrappedComponent
        useGroup
        userProfile={bidderUserObject}
      />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    const config = {
      onToastError: ['CzHHiJ2kw'],
      onToastInfo: ['CzHHiJ2kw'],
      onToastSuccess: ['CzHHiJ2kw'],
    };
    testDispatchFunctions(mapDispatchToProps, config);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<UserProfileGeneralInformation.WrappedComponent
      userProfile={bidderUserObject}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
