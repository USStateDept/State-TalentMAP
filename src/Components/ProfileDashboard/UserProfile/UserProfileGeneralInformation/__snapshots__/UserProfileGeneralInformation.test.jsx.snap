// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileGeneralInformationComponent matches snapshot 1`] = `
<div
  className="current-user-top current-user-section-container"
>
  <div
    className="section-padded-inner-container"
  >
    <ProfilePicture
      alt="avatar"
      className=""
      source="/assets/img/avatar.png"
    />
    <PositionInformation
      className="current-user-name"
      icon=""
      small={true}
      title="Doe, John"
    />
    <InformationDataPoint
      className="skill-code-data-point-container skill-code-data-point-container-skill"
      content={
        <SkillCodeList
          skillCodes={
            Array [
              Object {
                "code": "6218",
                "cone": null,
                "description": "CONSTRUCTION ENGINEERING",
                "id": 63,
              },
              Object {
                "code": "6211",
                "cone": null,
                "description": "COMPUTER SCIENCE",
                "id": 40,
              },
            ]
          }
        />
      }
      sideBySide={false}
      title=""
      titleOnBottom={false}
    />
  </div>
</div>
`;

exports[`UserProfileGeneralInformationComponent matches snapshot when showEditLink is true 1`] = `
<div
  className="current-user-top current-user-section-container"
>
  <div
    className="section-padded-inner-container"
  >
    <ProfilePicture
      alt="avatar"
      className=""
      source="/assets/img/avatar.png"
    />
    <PositionInformation
      className="current-user-name"
      icon=""
      small={true}
      title="Doe, John"
    />
    <InformationDataPoint
      className="skill-code-data-point-container skill-code-data-point-container-skill"
      content={
        <SkillCodeList
          skillCodes={
            Array [
              Object {
                "code": "6218",
                "cone": null,
                "description": "CONSTRUCTION ENGINEERING",
                "id": 63,
              },
              Object {
                "code": "6211",
                "cone": null,
                "description": "COMPUTER SCIENCE",
                "id": 40,
              },
            ]
          }
        />
      }
      sideBySide={false}
      title=""
      titleOnBottom={false}
    />
  </div>
</div>
`;

exports[`UserProfileGeneralInformationComponent matches snapshot when useGroup is true 1`] = `
<div
  className="current-user-top current-user-section-container"
>
  <div
    className="section-padded-inner-container"
  >
    <ProfilePicture
      alt="avatar"
      className=""
      source="/assets/img/avatar.png"
    />
    <PositionInformation
      className="current-user-name"
      icon=""
      small={true}
      title="Doe, John"
    />
    <Connect(StaticDevContent)>
      <InformationDataPoint
        className="skill-code-data-point-container skill-code-data-point-container-gen-spec"
        content="Generalist â€¢ F2"
        sideBySide={false}
        title=""
        titleOnBottom={false}
      />
    </Connect(StaticDevContent)>
  </div>
</div>
`;
